diff --git a/.gitignore b/.gitignore
index 2b1f44e..5e04a6b 100644
--- a/.gitignore
+++ b/.gitignore
@@ -2,7 +2,9 @@
 **/a.out
 
 # Ignore any executable without an extension (in any folder)
-**/[!.]*
+*
+!*/
+!*.*
 
 # Ignore all object files (in any folder)
 **/*.o
@@ -11,4 +13,4 @@
 **/core
 
 # Remove vscode config
-.vscode
\ No newline at end of file
+.vscode
diff --git a/test-gitignore/file123.txt b/test-gitignore/file123.txt
new file mode 100644
index 0000000..e69de29
diff --git a/test-gitignore/test1/abc2.txt b/test-gitignore/test1/abc2.txt
new file mode 100644
index 0000000..e69de29
diff --git a/test-gitignore/test2/test3/abc.txt b/test-gitignore/test2/test3/abc.txt
new file mode 100644
index 0000000..e69de29
